src/backend/joanie/lms_handler/edx_imports/edx_import.py

```diff
@@ -0,0 +1,334 @@
+            cleaned_data = lms.clean_course_run_data(edx_course_run_dict)
+            models.CourseRun.objects.filter(pk=target_course_run.pk).update(
+                **cleaned_data
+            )
+            course_run = models.CourseRun.objects.get(pk=target_course_run.pk)
+        else:
+            course_number = extract_course_number(edx_course_overview.id)
+            try:
+                course = models.Course.objects.get(code=course_number)
+            except models.Course.DoesNotExist:
+                course = models.Course.objects.create(
+                    created_on=format_date(edx_course_overview.created),
+                    code=course_number,
+                    title=edx_course_overview.display_name or course_number,
+                )
+                course.created_on = make_date_aware(edx_course_overview.created)
+                course.save()
+
+            # Instantiate a new course run
+            course_run = models.CourseRun.objects.create(
+                **edx_course_run_dict,
+                course=course,
+            )
+            course_run.created_on = make_date_aware(edx_course_overview.created)
+            course_run.save()
+
+        if title := edx_course_overview.display_name:
+            language_code = get_language_settings(
+                edx_course_overview.course.language
+            ).get("code")
+            course_run.set_current_language(language_code)
+            course_run.title = title
+        course_run.created_on = make_date_aware(edx_course_overview.created)
+        course_run.save()
+        logger.info("OK\n")
+    logger.info("Course runs import Done\n")
+
+
+def import_users(batch_size=1000):
+    """Import users from OpenEdx auth_user"""
+    logger.info("Getting users by batch of %s\n", batch_size)
+    db = OpenEdxDB()
+    users_count = db.get_users_count()
+
+    for current_user_index in range(0, users_count, batch_size):
+        start = current_user_index
+        stop = current_user_index + batch_size
+        start_time = time.time()
+        logger.info("  Processing %s-%s/%s ", start, stop, users_count)
+        users = db.get_users(start, stop)
+        get_time = time.time()
+        logger.info(" %s | ", f"{get_time - start_time:.2f}s")
+
+        build_time = time.time()
+        users_to_create = []
+        users_to_update = []
+
+        i = 0
+        for edx_user in users:
+            i += 1
+            if not i % 100:
+                logger.info(".")
+
+            if edx_user.username == "admin":
+                continue
+            last_login = format_date(edx_user.last_login)
+            date_joined = format_date(edx_user.date_joined)
+            try:
+                user = models.User.objects.get(username=edx_user.username)
+                user.email = edx_user.email
+                user.password = edx_user.password
+                user.first_name = edx_user.first_name
+                user.last_name = edx_user.last_name
```

Comment:
I'm afraid that those attributes will be always empty. If I remember well, instead we should retrieve the full name stored in the user Profile.

https://github.com/openfun/fonzie/blob/55a53e1d02d47af22125254bfa62006e7a7f65de/fonzie/views/user.py#L44